php_phongo.c

```diff
@@ -1921,6 +1921,17 @@ static bool php_phongo_apply_wc_options_to_uri(mongoc_uri_t* uri, bson_t* option
 } /* }}} */
 
 #ifdef MONGOC_ENABLE_SSL
+
+/* This is copied from mongoc-ssl.c */
+static void php_phongo_mongoc_ssl_opts_from_uri(mongoc_ssl_opt_t* ssl_opt, mongoc_uri_t* uri)
+{
+	ssl_opt->pem_file               = mongoc_uri_get_option_as_utf8(uri, MONGOC_URI_SSLCLIENTCERTIFICATEKEYFILE, NULL);
+	ssl_opt->pem_pwd                = mongoc_uri_get_option_as_utf8(uri, MONGOC_URI_SSLCLIENTCERTIFICATEKEYPASSWORD, NULL);
+	ssl_opt->ca_file                = mongoc_uri_get_option_as_utf8(uri, MONGOC_URI_SSLCERTIFICATEAUTHORITYFILE, NULL);
+	ssl_opt->weak_cert_validation   = mongoc_uri_get_option_as_bool(uri, MONGOC_URI_SSLALLOWINVALIDCERTIFICATES, false);
+	ssl_opt->allow_invalid_hostname = mongoc_uri_get_option_as_bool(uri, MONGOC_URI_SSLALLOWINVALIDHOSTNAMES, false);
```

Comment:
Ideas for new test cases:

 * Check that any single SSL-related URI option enables SSL for the client
 * Confirm that the `$driverOptions` equivalent overrides the corresponding URI option
 * Confirm that a separate `$driverOptions` key leaves an SSL-related URI option in place

Reply:
The original description in [PHPC-1239](https://jira.mongodb.org/browse/PHPC-1239) mentioned a URI with `ssl=true&sslallowinvalidcertificates=true`. Where did `sslallowinvalidcertificates=true` come from?

Assuming our MO environment still depends on self-signed certificates, I suppose we could rely on these undocumented URI options instead of passing the equivalent in `$driverOptions`. And the test could set an unrelated key in `$driverOptions` just to demonstrate that `sslallowinvalidcertificates=true` URI option is still respected.

We may not be able to use this approach for everything, but perhaps it can test one case, which would be better than none.